File: app_flutter/lib/models/article.dart
class Article {
  final int id;
  final String nom;
  final int quantiteALouer;
  final String? ref;
  final double? coefficient;
  final String? sau;

  Article({required this.id, required this.nom, required this.quantiteALouer, this.ref, this.coefficient, this.sau});

  factory Article.fromJson(Map<String, dynamic> json) => Article(
    id: json['id'] as int,
    nom: json['nom'] as String,
    quantiteALouer: json['quantiteALouer'] as int,
    ref: json['ref'] as String?,
    coefficient: (json['coefficient'] as num?)?.toDouble(),
    sau: json['sau'] as String?,
  );
}

File: app/src/main/java/com/quiz/mallzellij_show_android/model/Article.java
package com.quiz.mallzellij_show_android.model;

import java.math.BigDecimal;

public class Article {
    private Long id;
    private String nom;
    private Integer quantiteALouer;
    private String ref;
    private BigDecimal coefficient;
    private String sau;

    public Long getId() { return id; }
    public String getNom() { return nom; }
    public Integer getQuantiteALouer() { return quantiteALouer; }
    public String getRef() { return ref; }
    public BigDecimal getCoefficient() { return coefficient; }
    public String getSau() { return sau; }
}
